pkg/kn/core/root.go

```diff
@@ -340,7 +363,9 @@ func removeKnPluginFlags(args []string) []string {
 		if arg == "--plugins-dir" ||
 			strings.HasPrefix(arg, "--plugins-dir=") ||
 			arg == "--lookup-plugins" ||
-			strings.HasPrefix(arg, "--lookup-plugins=") {
+			strings.HasPrefix(arg, "--lookup-plugins=") ||
+			// remove -test.* args which are added when running go test
+			strings.HasPrefix(arg, "-test.") {
```

Comment:
Could you elaborate a bit on this ? I don't understand where this can happen.

Reply:
If you don't do this then running tests add these `-test.*` args and they fail the code during tests.